packages/runtime/test/lib/testUtils.ts

```diff
@@ -336,6 +340,109 @@ export async function establishRelationshipWithContents(
     expect(relationships2).toHaveLength(1);
 }
 
+export async function establishPendingRelationshipWithRequestFlow(
+    sRuntimeServices: TestRuntimeServices,
+    rRuntimeServices: TestRuntimeServices,
+    existingRelationshipAttributeForFurtherSharing?: LocalAttributeDTO
+): Promise<RelationshipDTO> {
```

Comment:
@britsta this is not _exactly_ how we discussed it. IMO this would be way more useful if you can throw in the whole items of the request and the items for the accept:

```ts
export async function establishPendingRelationshipWithRequestFlow(
    sRuntimeServices: TestRuntimeServices,
    rRuntimeServices: TestRuntimeServices,
    requestItems: (RequestItemGroupJSON | RequestItemJSONDerivations)[],
    acceptParams: (DecideRequestItemParametersJSON | DecideRequestItemGroupParametersJSON)[]
): Promise<RelationshipDTO> {
    const templateContent: RelationshipTemplateContentJSON = {
        "@type": "RelationshipTemplateContent",
        onNewRelationship: {
            "@type": "Request",
            items: requestItems
        }
    };

    const template = await exchangeTemplate(sRuntimeServices.transport, rRuntimeServices.transport, templateContent);

    await rRuntimeServices.eventBus.waitForEvent(IncomingRequestStatusChangedEvent, (e) => e.data.request.source!.reference === template.id);

    const requestId = (await rRuntimeServices.consumption.incomingRequests.getRequests({ query: { "source.reference": template.id } })).value[0].id;
    await rRuntimeServices.consumption.incomingRequests.accept({ requestId, items: acceptParams });

    await rRuntimeServices.eventBus.waitForEvent(RelationshipChangedEvent);

    const sRelationship = (await syncUntilHasRelationships(sRuntimeServices.transport, 1))[0];
    expect(sRelationship.status).toStrictEqual(RelationshipStatus.Pending);

    await sRuntimeServices.eventBus.waitForRunningEventHandlers();

    return sRelationship;
}
```

what do you think about this. In your case you would call this with 
```ts
establishPendingRelationshipWithRequestFlow(s1, s2, [
{
                        "@type": "CreateAttributeRequestItem",
                        mustBeAccepted: true,
                        attribute: IdentityAttribute.from({
                            value: {
                                "@type": "GivenName",
                                value: "AGivenName"
                            },
                            owner: (await rRuntimeServices.transport.account.getIdentityInfo()).value.address
                        }).toJSON()
                    },
                    {
                        "@type": "CreateAttributeRequestItem",
                        mustBeAccepted: true,
                        attribute: RelationshipAttribute.from({
                            key: "AKey",
                            value: {
                                "@type": "ProprietaryString",
                                value: "AStringValue",
                                title: "ATitle"
                            },
                            owner: CoreAddress.from((await rRuntimeServices.transport.account.getIdentityInfo()).value.address),
                            confidentiality: RelationshipAttributeConfidentiality.Public
                        }).toJSON()
                    },
                    {
                        "@type": "ShareAttributeRequestItem",
                        mustBeAccepted: true,
                        sourceAttributeId: existingRelationshipAttributeForFurtherSharing.id,
                        attribute: existingRelationshipAttributeForFurtherSharing.content
                    }
],
[{ accept: true }, { accept: true }, { accept: true }]
)
```

Reply:
I don't see us reusing exactly the same content as we have here - but if you want to have it re-usable you can put it in your test where you use it.